analytix/reports.py

```diff
@@ -105,15 +153,9 @@ async def ato_json(self, path: str, *, indent: int = 4) -> None:
                 with. Defaults to ``4``.
         """
 
-        if not path.endswith(".json"):
-            path += ".json"
-
-        async with aiofiles.open(path, "w") as f:
-            await f.write(json.dumps(self.data, indent=indent))
-
-        log.info(f"Saved report as JSON to {Path(path).resolve()}")
+        return self.to_json(path, indent=indent)
```

Comment:
Wouldn't this be `retuen await self.json(...)`?

Reply:
So, it can be - for some reason it makes it so that function returns the `DynamicReportWriter` instead of a `JSONReportWriter` when you await it inside. I'm not sure if you care about that though.

Reply:
probably because the await method is on the metaclass and is typed as returning `DynamicReportWriter`. This is fine.

Reply:
Yeah it doesn't really matter a great deal what it returns to be honest haha.